tests/integration/select2-methods.js

```diff
@@ -137,3 +137,87 @@ test('multiple value matches the jquery value', function (assert) {
     'The values should match the jquery values'
   );
 });
+
+test('selection and clearing of data from ajax source', function (assert) {
+  var asyncDone = assert.async();
+
+  var dataURL = 'http://127.0.0.1/test';
+  $.mockjax({
+    url: dataURL,
+    response: function (settings) {
+      this.responseText = {results: [{id: 6128, text: settings.data.term}]};
+    },
+    logging: 1
+  });
+
+  var $container = $('#qunit-fixture');
+  var $select = $('<select></select>');
+  $container.append($select);
+
+  var select = new Select2($select, {ajax: {url: dataURL}, multiple: true});
+
+  assert.equal(
+    $select.find(':selected').length,
+    0,
+    'No items should be selected'
+  );
+
+  var queryTerms = ['firstQuery', 'secondQuery', 'thirdQuery', 'fourthQuery'];
```

Comment:
This seems like a really smelly way to achieve multiple visits and actions on the `select2` control.  My JavaScript-fu isn't great, so there's almost certainly a better way to do this.